src/async/legends.ts

```diff
@@ -19,14 +19,29 @@ const layerLegendSelect = ({ layers, runConfiguration, job }: any) => {
 
 export default (store: any) => {
   // Layers legend
-  resync(store, layerLegendSelect, ({ layers, serviceId, objective, climate, region }, io, dispatch) => {
+  resync(store, layerLegendSelect, ({ layers }, io, dispatch) => {
     if (layers.length) {
-      const legendLayers = layers.filter((layer: any) => layer.displayed === true && layer.type !== 'vector')
+      const legendLayers = layers.filter((layer: any) =>
+        typeof config.layers[layer].show === 'boolean'
+          ? config.layers[layer].show
+          : config.layers[layer].show(store.getState()) && config.layers[layer].type !== 'vector',
+      )
+
       dispatch(resetLegends())
-      legendLayers.forEach((layer: any) => {
+      legendLayers.forEach((layer: string) => {
         dispatch(requestLayersLegend())
-        const newrl = getLayerUrl(layer, serviceId, objective, climate, region)
-        const url = `/arcgis/rest/services/${newrl}/MapServer/legend`
+        const { legendUrl } = config.layers[layer]
+        let url
+        switch (typeof legendUrl) {
+          case 'undefined':
+            url = ''
```

Comment:
When `url = ''`, we shouldn't try to fetch the layer below (`return io.get(...)`), since that will _always_ be an error.

Reply:
I changed this to just return when `url = ''`